sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/PartitionSynchronizerImpl.java

```diff
@@ -93,11 +88,23 @@ public PartitionSynchronizerImpl(
         // or at least Service - this will be part of the next set of changes
         // For now - no merge just simple V0 of lease contract
         // this simplification will work
-        final String lastContinuationToken = lastContinuationState.getContinuation() != null ?
-            lastContinuationState.getContinuation().getCurrentContinuationToken().getToken() :
-            null;
-
-        logger.info("Partition {} is gone due to split.", leaseToken);
+        //
+        //ChangeFeedState lastContinuationState = lease.getContinuationState(
+        //    this.collectionResourceId,
+        //    new FeedRangePartitionKeyRangeImpl(leaseToken)
+        //);
+        //
+        //final String lastContinuationToken = lastContinuationState.getContinuation() != null ?
+        //    lastContinuationState.getContinuation().getCurrentContinuationToken().getToken() :
+        //    null;
+
+        // "Push" ChangeFeedProcessor is not merge-proof currently. For such cases we need a specific handler that can
+        // take multiple leases and "converge" them in a thread safe manner while also merging the various continuation
+        // tokens for each merged lease.
+        // We will directly reuse the original/parent continuation token as the seed for the new leases until then.
+        final String lastContinuationToken = lease.getContinuationToken();
```

Comment:
Add tests

Reply:
@j82w We have tests that ensure CFP properly recovers while split occurs. However because the nature of how CFP works, receiving duplicates is expected and trying to time exactly when to look for the state of the current continuation tokens in the leases (which btw are internal and not directly accessible), it will potentially add flakiness in our CI.
I will continue to explore how to add a potential test but given the urgency of this fix, this is not a priority at the moment.

Reply:
We have similar test scenarios for partition split in the rest of the SDK components, the following can be used as example for adding tests in CFP for partition split:
1. mocking unit test: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/query/DocumentProducerTest.java#L110
2. end to end test: 
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos-benchmark/src/test/java/com/azure/cosmos/benchmark/ReadMyWritesConsistencyTest.java

Reply:
The main issue is not about not having a split test, we already have that. The bug is about the content of the resulting leases after the split, which is information that is actively updated and not directly exposed to the test as a set of public APIs.
We can use the "BridgeInternals" to directly access the internal CFP classes and call in directly into the methods we want to unit test.
Alternatively and the approach we expect to mirror customer scenarios is to use the public APIs provided and the CFP to monitor itself, in this case the lease collection. Any updates to that collection can be monitored and validated real time. The only complication with this approach is that timing for the various checks can be thrown off depending on connectivity, thread scheduling and other conditions. In the current test we introduce enough of a lag between the various checks to account for any such situations.